File: 2022/day03/main.py
priorities = 'abcdefghijklmnopqrstuvwxyzABCDEFGHIJKLMNOPQRSTUVWXYZ'

def solve1(rucksacks):
    sum = 0

    for rucksack in rucksacks:
        c1, c2 = rucksack[:len(rucksack)//2], rucksack[len(rucksack)//2:]

        common = set(c1).intersection(c2)

        sum += priorities.find(common.pop()) + 1

    return sum

def solve2(rucksacks):
    sum = 0
    common = set()

    for rucksack in rucksacks:
        if len(common) == 0:
            common = set(rucksack)
        else:
            common = common.intersection(rucksack)

            if len(common) == 1:
                sum += priorities.find(common.pop()) + 1

    return sum


def main():
    rucksacks = [line.rstrip() for line in open("input.txt")]

    print(solve1(rucksacks))
    print(solve2(rucksacks))


if __name__ == "__main__":
    main()

File: 2022/day03/src/main.rs
use std::fs::File;
use std::collections::HashSet;
use std::io::{self, BufReader, BufRead};

fn read_input(file_name: &str) -> io::Result<BufReader<File>> {
    let file = File::open(file_name)?;
    Ok(BufReader::new(file))
}

fn solve2(input: BufReader<File>) -> i32 {
    let mut sum = 0;
    let mut common = HashSet::new();

    for line in input.lines() {
        if common.len() == 0 {
            for character in line.unwrap().chars() {
                common.insert(character);
            }
        } else {
            
        }
    }
        

    0
}

fn main() {
    let input: BufReader<File> = read_input("input.txt").unwrap();
    
    println!("Part 2: {}", solve2(input));
}